channel/handlers_client.go

```diff
@@ -448,8 +448,115 @@ func (server Server) playerEmote(conn mnet.Client, reader mpacket.Reader) {
 }
 
 func (server Server) playerUseMysticDoor(conn mnet.Client, reader mpacket.Reader) {
-	// doorID := reader.ReadInt32()
-	// fromTown := reader.ReadBool()
+	plr, err := server.players.GetFromConn(conn)
+	if err != nil {
+		return
+	}
+
+	if plr.inst == nil {
+		plr.Send(packetPlayerNoChange())
+		return
+	}
+
+	var (
+		nearestDoor *mysticDoorInfo
+		ownerID     int32
+		minDist     int32 = 1<<30 - 1
+	)
+	for oid, door := range plr.inst.mysticDoors {
```

Comment:
Could attach mystic doors to players and then iterate over party players instead of the whole instance